Implement Signing HTTP Messages with structured field values (#47)
digitalbazaar/http-signature-header: README.md

//sig1=("one");alg="foo"
```

example: 
```js
const dict = httpSignatureHeader.createSignatureInputHeader({
  // here we have 2 signatures for the request
  signatures: new Map([
    ['sig1', {coveredContent: [{value: 'one'}]}],
    ['sig2', {coveredContent: [{value: 'two'}]}]
  ]),
  // this will add the params to all the signatures
  params: {alg: 'foo'}
});
// this is a valid `Signature-Input` header:
console.log(dict);
//sig1=("one");alg="foo", sig2=("two");alg="foo"
```
example: 
```js
const dict = httpSignatureHeader.createSignatureInputHeader({
  // here we have 2 signatures for the request
  signatures: new Map([
    ['sig1', {coveredContent: [{value: 'one'}], params: {alg: 'foo'}}],
  // sig2 uses a different signing algorithm `bar`
    ['sig2', {coveredContent: [{value: 'two'}], params: {alg: 'bar'}}]
  ]),
});
// this is a valid `Signature-Input` header:
console.log(dict);
//sig1=("one");alg="foo", sig2=("two");alg="bar"
```

`createSignatureInputString`

```js
const date = new Date().toUTCString();
const httpMessage = {
  headers: {date},
  method: 'GET',
  url: 'https://example.com',
};
const coveredContent = [{value: 'one'}];
const params = {alg: 'foo'};

> // this turns your coveredContent etc into SF Items

Reviewer: ```suggestion
// this turns the coveredContent etc into SF Items
```
Also, `SF` is used here and throughout (variously as `sf`) without introducing the term.  For my part, I'd prefer that it was 'structured field' throughout.

const stringToSign = httpSignatureHeader.createSignatureInputString({
  coveredContent,
  params,
  httpMessage
});
const data = new TextEncoder().encode(stringToSign);
// invocationSigner being a key
const signature = await invocationSigner.sign({data});
```

> `createSignatureHeader` is much easier. Sign the signing string for your signature
> and pass it as a `Uint8Array`.

Reviewer: ```suggestion
The `createSignatureHeader` API is used to encode the signature bytes provided as a `Uint8Array`.
```

example: 
```js
const signature = new Uint8Array([1, 2, 3])
const sigs = {sig1: signature};
const dict = httpSignatureHeader.createSignatureHeader(sigs);
console.log(dict);
//'sig1=:AQID:'
```

> `parseRequest` takes in a request that is expected to be similar to
> a request object in `express.js`. It then parses the `Signature-Input` header and returns
> a Map with the signature id and then an object with everything you need to sign:

Reviewer: ```suggestion
The `parseRequest` API takes in a request object similar to a request object in `express.js`. 
It then parses the `Signature-Input` header and returns
a Map with the signature id and then an object with everything needed to create a signed header:
```
I was confused by the meaning of "to sign", I changed it above to what I think is meant by those words.
